### ISSUE TITLE
Lwt lazy value exploding and causing early exit

### ISSUE THREAD

**Issue by fxfactorial** [fxfactorial/ocaml-usbmux#3]
```
Mar 23 22:12:38 conquest gandalf[21356]: main: Exited with 19 still running; this is a bug.
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: Called from file "src/core/lwt.ml", line 781, characters 66-71
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: Called from file "src/core/lwt.ml", line 1150, characters 14-18
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: Called from file "src/unix/lwt_io.ml", line 1387, characters 28-36
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: Re-raised at file "camlinternalLazy.ml", line 32, characters 10-11
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: Called from file "camlinternalLazy.ml", line 25, characters 17-27
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: Raised at file "camlinternalLazy.ml", line 18, characters 48-57
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: backtrace:
Mar 23 22:12:38 conquest gandalf[21356]: async_exceptions: Please report, this is an unhandled async exception (A bug): CamlinternalLazy.Undefined
```

I think its an issue of 

``` ocaml
        (* Finished tunneling, now cleanly close the chans *)
        >>= close_chans (tcp_ic, tcp_oc)
        |> Lwt.ignore_result
```

in `usbmux.ml` because lwt's `establish_server` has its own logic on closing the server's connections and uses `lazy` in its implementation.

**Comment by fxfactorial** [comment]
This is possibly fixed in c1b4d2909ac5f62f2a43f73b6833a04c2e80501d testing now

**Comment by fxfactorial** [comment]
Fixed.
